src/prng.rs

```diff
@@ -61,74 +119,42 @@ pub enum PrngError {
 /// This type implements an iterator that generates a pseudorandom sequence of field elements. The
 /// sequence is derived from the key stream of AES-128 in CTR mode with a random IV.
 #[derive(Debug)]
-pub(crate) struct Prng<F: FieldElement> {
+pub(crate) struct Prng<F: FieldElement, X: StreamCipher> {
     phantom: PhantomData<F>,
-    cipher: Aes128Ctr,
-    length: Option<usize>,
+    key_stream: X,
     buffer: Vec<u8>,
     buffer_index: usize,
     output_written: usize,
 }
 
-impl<F: FieldElement> Prng<F> {
+impl<F: FieldElement, X: StreamCipher> Prng<F, X> {
     /// Generates a seed and constructs an iterator over an infinite sequence of pseudorandom field
     /// elements.
-    pub(crate) fn new() -> Result<Self, getrandom::Error> {
-        let mut seed = [0; SEED_LENGTH];
+    pub(crate) fn new() -> Result<Self, PrngError> {
+        let mut seed = vec![0; X::STREAM_CIPHER_SEED_SIZE];
         getrandom(&mut seed)?;
-        Ok(Self::new_with_seed_and_optional_length(&seed, None))
-    }
-
-    /// Generates a seed and constructs an iterator over a pseudorandom sequence of field elements
-    /// of length `length`.
-    pub(crate) fn new_with_length(length: usize) -> Result<Self, getrandom::Error> {
-        let mut seed = [0; SEED_LENGTH];
-        getrandom(&mut seed)?;
-        Ok(Self::new_with_seed_and_optional_length(&seed, Some(length)))
-    }
-
-    /// Creates a PRNG from a seed.
-    pub(crate) fn try_from_seed(seed: &[u8]) -> Result<Self, PrngError> {
-        if seed.len() != SEED_LENGTH {
-            return Err(PrngError::SeedLen);
-        }
-
-        Ok(Self::new_with_seed_and_optional_length(seed, None))
+        Self::new_with_seed(&seed)
     }
 
-    fn new_with_seed_and_optional_length(seed: &[u8], length: Option<usize>) -> Self {
-        let key = GenericArray::from_slice(&seed[..BLOCK_SIZE]);
-        let iv = GenericArray::from_slice(&seed[BLOCK_SIZE..]);
-        let mut cipher = Aes128Ctr::from_block_cipher(Aes128::new(key), iv);
-
-        let buf_len_in_elems = match length {
-            Some(length) => std::cmp::min(length + 1, MAXIMUM_BUFFER_SIZE_IN_ELEMENTS),
-            None => DEFAULT_BUFFER_SIZE_IN_ELEMENTS,
-        };
-        let mut buffer = vec![0; buf_len_in_elems * F::ENCODED_SIZE];
-        cipher.apply_keystream(&mut buffer);
+    pub(crate) fn new_with_seed(seed: &[u8]) -> Result<Self, PrngError> {
+        let mut key_stream = X::new(seed)?;
+        let mut buffer = vec![0; BUFFER_SIZE_IN_ELEMENTS * F::ENCODED_SIZE];
```

Comment:
nit: since the size of the buffer depends upon constants, `Prng.buffer` could be a byte slice... doesn't need to be done in this PR though.

Reply:
:(
```
error: generic parameters may not be used in const operations
   --> src/prng.rs:126:44
    |
126 |     buffer: [u8; BUFFER_SIZE_IN_ELEMENTS * F::ENCODED_SIZE],
    |                                            ^^^^^^^^^^^^^^^ cannot perform const operation using `F`
    |
    = note: type parameters may not be used in const expressions
```